src/core/orderbookutils.js

```diff
@@ -50,7 +85,10 @@ OrderBookUtils.getOfferTakerPaysFunded = function(offer) {
 OrderBookUtils.getOfferTakerGets = function(offer) {
   assert(typeof offer, 'object', 'Offer is invalid');
 
-  return Amount.from_json(offer.TakerGets + IOU_SUFFIX);
+  let currency = offer.TakerPays.currency;
+  let issuer = offer.TakerPays.issuer;
+
+  return Amount.from_json(createAmount(offer.TakerGets, currency, issuer));
```

Comment:
Could be either a string OR an object amount, assuming it's in rippled format, which the capitalisation of the word indicates. It seems this (seeming) bug existed prior to this patch. Note the diff:
`- return Amount.from_json(offer.TakerGets + IOU_SUFFIX);`

It seems like the util is used in `autobridgecalculator`, which is code I'm as yet unacquainted with.